packages/design-preview/src/components/index.tsx

```diff
@@ -1,17 +1,43 @@
+import { useEffect, useState } from '@wordpress/element';
 import Sidebar from './sidebar';
-import type { StyleVariation } from '@automattic/design-picker';
+import SitePreview from './site-preview';
+import type { StyleVariation } from '@automattic/design-picker/src/types';
 import './style.scss';
 
 interface PreviewProps {
+	previewUrl: string;
 	title?: string;
 	description?: string;
 	variations?: StyleVariation[];
 }
 
-const Preview: React.FC< PreviewProps > = ( { title, description, variations = [] } ) => {
+const Preview: React.FC< PreviewProps > = ( {
+	previewUrl,
+	title,
+	description,
+	variations = [],
+} ) => {
+	const [ activeVariation, setActiveVariation ] = useState< StyleVariation | undefined >();
+	const handleVariationClick = ( variation: StyleVariation ) => {
+		setActiveVariation( variation );
+	};
+
+	useEffect( () => {
```

Comment:
Can we instead simply use `useState( variations[0] )` above ?

Reply:
Oh, I think I added the `useEffect` because  `variations[0]` would be set to empty while fetch from the API if I used `useState`. Maybe I'm missing something? 🤔

Reply:
Oops, sorry you're right. Then I think we need to change the below condition to 

```
if ( variations.length > 0 && ! activeVariations )
```

So that `variations[0]` is accessed when it is not empty?